webpack/assets/javascripts/react_app/components/factCharts/factChart.test.js

```diff
@@ -0,0 +1,30 @@
+import { shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import React from 'react';
+import ConnectedFactChart, { FactChart } from './';
+import Store from '../../redux';
+
+const state = {
+  chartData: [],
+  loaderStatus: '',
+};
+
+describe('factCharts', () => {
+  it('should render closed', () => {
+    const wrapper = shallow(<FactChart data={ { id: 1, title: 'test title' } } factChart={state}/>);
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should render open', () => {
+    const wrapper = shallow(<FactChart data={ { id: 1, title: 'test title' } } factChart={state} modalToDisplay />);
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should open modal after click', () => {
+    const props = { onClick: jest.fn(), data: { id: 1 } };
+    const wrapper = shallow(<ConnectedFactChart store={ Store } { ...props } />);
+
+    wrapper.simulate('click');
+    expect(props.onClick).toBeCalled();
+  });
```

Comment:
missing tests for searching?

Reply:
shouldn't this functionality be tested at `ChartService`?